src/shared/qmlpath.h

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <QObject>
+
+class QQmlApplicationEngine;
+class QQuickItem;
+
+/**
+ * @brief Filter a QML tree using an XPath-like syntax
+ *
+ * The syntax is composed of blocks of "path". A path can be one of the
+ * following:
+ *
+ * - '/' -> all the items at the current QML tree level
+ * - '/' + objectName - the item with the name `objectName` as a child of the
+ *   current QML tree item
+ * - '//' + objectName - search of an item with the name 'objectName'
+ *   recursively from the current QML tree item
+ *
+ * You can apply 0, 1, or more filters to paths.
+ * - '[' + number + ']' -> from all the items selected from the path block,
+ *   pick the one in position `number`.
+ * - '{' + propertyName + '}' - from all the items selected from the path
+ *   block, pick the ones that have a property with the name `propertyName`.
+ * - '{' + propertyName + '=' + propertyValue + '}' - from all the items
+ *   selected from the path block, pick the ones with the property with the name
+ *   `propertyName` and value set to `propertyValue`.
+ *
+ * Paths blocks can be concatenated: `/abc//foo[1]{p=42}/bar`
+ *
+ * Additionally, you can run a command to the final path with the syntax:
```

Comment:
I am a BIG fan of this x-path style, **but**. 
Having statements to both query- and change state, that sounds too powerful - lot's of concerns here. 
I'd prefer this being restricted to output ref's that you can pass to an operator-function. 
This would allow us to make sure the context when Action X on Ref Y is performed is correct (see addon-manifest-question). 
Also there does not seem to be any user of the command-style right now, imho we should upgrade the old inspector actions to use QmlPaths instead of ::findName,

Reply:
I agree with you. Let's drop that part.